In [1]:
import numpy as np
import pandas as pd
import scipy .integrate as integrate
import scipy
import matplotlib.pyplot as plt
from scipy.special import gamma as gamma 

## 
$$
h(t; \nu) = \frac{\Gamma[(\nu + 1)/2]}{\Gamma(\nu/2)\sqrt{\pi\nu}}\left (1 + \frac{t^2}{\nu} \right)^{-(\nu + 1)/2}
$$

In [3]:
def tdist (t, nu):
    return gamma((nu + 1)*0.5)/(gamma(nu*0.5)*np.sqrt(np.pi*nu))*(1.0 + t**2/nu)**(-(nu + 1)*0.5)

In [5]:
x_data = [9.8, 10.2, 10.4, 9.8, 10.0, 10.2, 9.6]
x_mean = 10
n = 7
nu = n - 1
alpha =0.05
alpha2 = 0.025

In [7]:
np.mean(x_data)

np.float64(10.0)

In [9]:
s2 = np.var(x_data, ddof = 1)
print('desv. Estandar muestral', np.sqrt(s2))

desv. Estandar muestral 0.28284271247461884


In [ ]:
nu = 14
x_array = np.linspace(-5, 5, 1000)
y_array = tdist(x_array, nu)
plt.plot(x_array, y_array, linestyle = '-', color = 'red', label = r'$\nu = %d$' %nu)
plt.legend(fontsize = 14)
plt.show()

Calculo del valor de $t_{0.025}$

In [11]:
t_array = np.linspace(-10, 10, 1000)

ss = 0
for tval in t_array:
    prob = integrate.quad(tdist, -np.inf, tval, args = (nu))
    if prob[0] >= 0.975:
        print(t_array[ss-1])
        break
    ss+=1

2.4324324324324316


Entonces, $$ t_{0.025} = 2.43$$

El intervalo de confianza para $\mu$ con $\sigma$ Descnocido es: